src/libstore/derivations.hh

```diff
@@ -13,10 +13,20 @@ namespace nix {
 
 /* Abstract syntax of derivations. */
 
-struct DerivationOutput
+struct DerivationOutputIntensional
 {
     StorePath path;
-    std::optional<FixedOutputHash> hash; /* hash used for expected hash computation */
+};
+
+struct DerivationOutputFixed
```

Comment:
Oh, it just occurred to me I guess we could get rid of this and just use `FixedOutputHash` directly.

Reply:
yeah, but it would be less type safe right? we could do the same with DerivationOutputExtensional and StorePath

Reply:
I wouldn't say "less type safe", just a matter of preference. I don't have a preference, just pointing out the possibility for others that might.